### PR TITLE
:sparkles: Add assessment/review status to archetype table

### PR DESCRIPTION
**Pull request by rszwajko**
Review states:
1. Completed - a review exists
2. NotStarted

Assessment states:
1. Completed - all required assessments done (based on 'assessed' flag)
2. InProgress - some assessments done
3. NotStarted

Resolves: https://github.com/konveyor/tackle2-ui/issues/1751

### CODE REVIEW

**Comment by rszwajko** [comment]
### Assessment completed, review not started
Screenshot from 2024-03-08 18-53-46
### Assessment in progress, review completed
Screenshot from 2024-03-08 18-48-31

**Review by ibolton336 — DISMISSED** [review]
Look good so far. Discarding review/assessment doesn't invalidate the archetypes query & results in stale data.

**Comment by sjd78** [comment]
> Look good so far. Discarding review/assessment doesn't invalidate the archetypes query & results in stale data.

This is what application table does:

https://github.com/konveyor/tackle2-ui/blob/3e3d6904964fd4085bdfddd3e7530368a0022b10/client/src/app/pages/applications/applications-table/applications-table.tsx#L246-L258

https://github.com/konveyor/tackle2-ui/blob/3e3d6904964fd4085bdfddd3e7530368a0022b10/client/src/app/pages/applications/applications-table/applications-table.tsx#L291-L306

I'm not a fan of the invalidation in the component's mutation onSuccess() handler, but it is what it is.

File: client/src/app/pages/archetypes/archetypes-page.tsx [hunk 1]
```diff
@@ -154,25 +162,29 @@ const Archetypes: React.FC = () => {
...
-          id: archetype.review.id,
-          name: archetype.name,
+    if (!archetype.review?.id) {
+      return;
+    }
+    deleteReview({
+      id: archetype.review.id,
+      name: archetype.name,
+    })
+      .then(() => {
```
Comment (sjd78) [inline client/src/app/pages/archetypes/archetypes-page.tsx:177]: Nit-pick: With the promise callback style, the function doesn't need to be `async`.

File: client/src/app/pages/archetypes/archetypes-page.tsx [hunk 2]
```diff
@@ -115,34 +120,37 @@ const Archetypes: React.FC = () => {
...
+    }
+    Promise.all(
+      archetype.assessments.map(async (assessment) => {
+        await deleteAssessment({
+          assessmentId: assessment.id,
+          archetypeId: archetype.id,
+        });
+      })
+    )
+      .then(() => {
```
Comment (sjd78) [inline client/src/app/pages/archetypes/archetypes-page.tsx:137]: Nit-pick: With the promise callback style, the function doesn't need to be `async`.

**Review by sjd78 — APPROVED** [review]
LGTM

Tested and the assessment and review columns seem to reset immediately after the delete/discard is complete.  Looks like the query invalidations are working correctly now.

**Comment by rszwajko** [comment]
@sjd78 removed unnecessary `async` in the newest force push

**Review by sjd78 — APPROVED** [review]
LGTM and no more nit-picks!  :-)

**Review by ibolton336 — APPROVED** [review]
LGTM!

**Review by aufi — APPROVED** [review]
LGTM

**Pull request merged**